utils/errors/errors.go

```diff
@@ -145,26 +155,31 @@ func NewWithSkip(msg string, skip int) ErrorExt {
 
 //NewWithSkipAndStatus creates a new error skipping the number of function on the stack and GRPC status
 func NewWithSkipAndStatus(msg string, skip int, status *grpcstatus.Status) ErrorExt {
-	return WrapWithSkipAndStatus(fmt.Errorf(msg), "", skip+1, status)
+	return wrapWithSkipAndStatus(fmt.Errorf(msg), "", skip+1, status)
 }
 
 //Wrap wraps an existing error and appends stack information if it does not exists
 func Wrap(err error, msg string) ErrorExt {
 	return WrapWithSkip(err, msg, 1)
 }
 
-//Wrap wraps an existing error and appends stack information if it does not exists along with GRPC status
+//WrapWithRawData wraps an existing error and appends stack information if it does not exists
+func WrapWithRawData(err error, msg string, rawData ...interface{}) ErrorExt {
+	return wrapWithSkipAndStatus(err, msg, 2, nil, rawData)
+}
+
+//WrapWithStatus wraps an existing error and appends stack information if it does not exists along with GRPC status
 func WrapWithStatus(err error, msg string, status *grpcstatus.Status) ErrorExt {
-	return WrapWithSkipAndStatus(err, msg, 1, status)
+	return wrapWithSkipAndStatus(err, msg, 1, status)
 }
 
 //WrapWithSkip wraps an existing error and appends stack information if it does not exists skipping the number of function on the stack
 func WrapWithSkip(err error, msg string, skip int) ErrorExt {
-	return WrapWithSkipAndStatus(err, msg, skip+1, nil)
+	return wrapWithSkipAndStatus(err, msg, skip+1, nil)
 }
 
-//WrapWithSkip wraps an existing error and appends stack information if it does not exists skipping the number of function on the stack along with GRPC status
-func WrapWithSkipAndStatus(err error, msg string, skip int, status *grpcstatus.Status) ErrorExt {
+//wrapWithSkipAndStatus wraps an existing error and appends stack information if it does not exists skipping the number of function on the stack along with GRPC status
+func wrapWithSkipAndStatus(err error, msg string, skip int, status *grpcstatus.Status, rawData ...interface{}) ErrorExt {
```

Comment:
don't expose this real logic func

Reply:
I've checked no one using it

Reply:
Users may defined their own errors on top of Orion. If these function are not exposed, then the call stacks for functions that generate the error will not be skipped. Since one can only call Wrap function but it always skips starting from Wrap but not the wrapping function users defined.

Reply:
sure, it make sense to me.